signer/src/testing/context.rs

```diff
@@ -206,6 +206,43 @@ impl<Storage, Bitcoin, Stacks>
     }
 }
 
+/// Addition to Context trait, used in tests.
+pub trait TestingContext: Clone + Sync + Send {
```

Comment:
Why was this needed? If you need to modify settings on-the-fly with the `TestContext` below you can use `.modify_settings(|config| { ... })`.

Reply:
`modify_settings` works for `TestContext::builder()`, while I want sometimes change settings for already built context

Reply:
[634b84b](https://github.com/stacks-sbtc/sbtc/pull/1602/commits/634b84beeb03c4dfd8a64efec19597c1e80a8eb8)